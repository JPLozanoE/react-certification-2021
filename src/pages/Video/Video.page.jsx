import React from 'react';
import { VideoPlayerView } from '../../components/Home/VideoPlayerView';

export const VideoPage = () => {
  return <VideoPlayerView favorites={false} />;
};
