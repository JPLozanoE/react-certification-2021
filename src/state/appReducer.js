import { types } from '../types/types';

export const appReducer = (state = {}, action) => {
  switch (action?.type) {
    case types.changeSearchQuery:
      return {
        ...state,
        search: action.payload,
      };

    case types.setSelectedVideo:
      return {
        ...state,
        selectedVideo: action.payload,
      };

    case types.toggleTheme:
      return {
        ...state,
        isDarkTheme: !state.isDarkTheme,
      };

    case types.setRecommendedVideos:
      return {
        ...state,
        videos: action.payload,
      };

    case types.login:
      return {
        ...state,
        auth: {
          isLogged: true,
          displayName: action.payload.displayName,
          uid: action.payload.uid,
        },
      };

    case types.logout:
      return {
        ...state,
        auth: {
          isLogged: false,
          displayName: '',
          uid: null,
        },
      };

    case types.addFavoriteVideo:
      return {
        ...state,
        favoriteVideos: [
          ...state.favoriteVideos,
          { ...action.payload, isFavorite: true },
        ],
      };

    case types.deleteFavoriteVideo:
      return {
        ...state,
        videos: state.videos.map((video) =>
          video.id.videoId === action.payload ? { ...video, isFavorite: false } : video
        ),
        favoriteVideos: state.favoriteVideos.filter(
          (favoriteVideo) => favoriteVideo.id.videoId !== action.payload
        ),
      };

    case types.updateRecommendedVideos:
      return {
        ...state,
        videos: state.videos.map((video) =>
          video.id.videoId === action.payload ? { ...video, isFavorite: true } : video
        ),
      };

    default:
      return state;
  }
};
