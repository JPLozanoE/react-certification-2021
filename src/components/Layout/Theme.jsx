import React, { useContext } from 'react';
import { ThemeProvider } from 'styled-components';
import { AppContext } from '../../state/AppContext';

const Theme = ({ children }) => {
  const {
    state: { isDarkTheme },
  } = useContext(AppContext);
  const theme = {
    main: {
      ...(isDarkTheme ? { backgroundColor: '#181818' } : { backgroundColor: '#ffffff' }),
    },
    titles: {
      ...(isDarkTheme ? { color: '#ffffff' } : { color: '#00000' }),
    },
    subtitles: {
      ...(isDarkTheme ? { color: 'rgb(170,170,170)' } : { color: 'rgb(96,96,96)' }),
    },
  };

  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

export default Theme;
