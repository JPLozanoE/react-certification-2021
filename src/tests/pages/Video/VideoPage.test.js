/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { AppContext } from '../../../state/AppContext';
import Theme from '../../../components/Layout/Theme';
import { demoState, singleVideo } from '../../fixtures/demoState';
import { VideoPage } from '../../../pages/Video/Video.page';

describe('Video Page', () => {
  let wrapper;

  test('Should render properly', () => {
    const providerValues = {
      state: { ...demoState, selectedVideo: singleVideo },
      dispatch: jest.fn(),
    };
    wrapper = render(
      <AppContext.Provider value={providerValues}>
        <Theme>
          <Router>
            <VideoPage />
          </Router>
        </Theme>
      </AppContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.getByText(demoState.videos[0].snippet.title)).toBeInTheDocument();
  });
});
