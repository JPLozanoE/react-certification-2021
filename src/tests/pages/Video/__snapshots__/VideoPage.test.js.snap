// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video Page Should render properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdnxRM gbwLTQ"
      >
        <div
          class="sc-gtsrHT gBukoy"
        >
          <iframe
            allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
            allowfullscreen=""
            class="sc-eCApnc cepYYm"
            frameborder="0"
            src="https://www.youtube.com/embed/1-2eyS0k-IA"
            title="Embedded youtube"
            width="100%"
          />
          <h2
            class="sc-dlnjwi dzceGW"
          >
            ¿Qué es React JS? Breve explicación animada
          </h2>
          <p
            class="sc-hKFxyN cvlDaB"
          >
            React (también llamada React.js o ReactJS) es una librería Javascript de código abierto para crear interfaces de usuario con el objetivo de animar al desarrollo ...
          </p>
        </div>
        <div
          class="sc-gtsrHT gBukoy"
        >
          <div
            class="sc-jrsJWt cwylym"
          >
            <div
              class="sc-fujyAs ddSJEO"
            >
              <img
                alt="img"
                aria-label="Twins vs. Yankees Game Highlights (8/20/21) | MLB Highlights"
                src="https://i.ytimg.com/vi/yFq-c9ZpmRs/default.jpg"
              />
              <div
                class="sc-pNWdM kVAvoZ"
              >
                Twins vs. Yankees Game Highlights (8/20/21) | MLB Highlights
                <div>
                  <button
                    class="sc-gKAaRy kTkCxO"
                    type="button"
                  >
                    Add to Favorites
                  </button>
                </div>
              </div>
            </div>
            <hr />
            <div
              class="sc-fujyAs ddSJEO"
            >
              <img
                alt="img"
                aria-label="State of Emergency As Hurricane Henri Approaches New York"
                src="https://i.ytimg.com/vi/EqG9gWbwxlw/default.jpg"
              />
              <div
                class="sc-pNWdM kVAvoZ"
              >
                State of Emergency As Hurricane Henri Approaches New York
                <div>
                  <button
                    class="sc-gKAaRy kTkCxO"
                    type="button"
                  >
                    Add to Favorites
                  </button>
                </div>
              </div>
            </div>
            <hr />
            <div
              class="sc-fujyAs ddSJEO"
            >
              <img
                alt="img"
                aria-label="MAGICIAN BRANDS HIS OWN FLESH... WITH A PLAYING CARD!"
                src="https://i.ytimg.com/vi/LqiDJh4zxAc/default.jpg"
              />
              <div
                class="sc-pNWdM kVAvoZ"
              >
                MAGICIAN BRANDS HIS OWN FLESH... WITH A PLAYING CARD!
                <div>
                  <button
                    class="sc-gKAaRy kTkCxO"
                    type="button"
                  >
                    Add to Favorites
                  </button>
                </div>
              </div>
            </div>
            <hr />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdnxRM gbwLTQ"
    >
      <div
        class="sc-gtsrHT gBukoy"
      >
        <iframe
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
          allowfullscreen=""
          class="sc-eCApnc cepYYm"
          frameborder="0"
          src="https://www.youtube.com/embed/1-2eyS0k-IA"
          title="Embedded youtube"
          width="100%"
        />
        <h2
          class="sc-dlnjwi dzceGW"
        >
          ¿Qué es React JS? Breve explicación animada
        </h2>
        <p
          class="sc-hKFxyN cvlDaB"
        >
          React (también llamada React.js o ReactJS) es una librería Javascript de código abierto para crear interfaces de usuario con el objetivo de animar al desarrollo ...
        </p>
      </div>
      <div
        class="sc-gtsrHT gBukoy"
      >
        <div
          class="sc-jrsJWt cwylym"
        >
          <div
            class="sc-fujyAs ddSJEO"
          >
            <img
              alt="img"
              aria-label="Twins vs. Yankees Game Highlights (8/20/21) | MLB Highlights"
              src="https://i.ytimg.com/vi/yFq-c9ZpmRs/default.jpg"
            />
            <div
              class="sc-pNWdM kVAvoZ"
            >
              Twins vs. Yankees Game Highlights (8/20/21) | MLB Highlights
              <div>
                <button
                  class="sc-gKAaRy kTkCxO"
                  type="button"
                >
                  Add to Favorites
                </button>
              </div>
            </div>
          </div>
          <hr />
          <div
            class="sc-fujyAs ddSJEO"
          >
            <img
              alt="img"
              aria-label="State of Emergency As Hurricane Henri Approaches New York"
              src="https://i.ytimg.com/vi/EqG9gWbwxlw/default.jpg"
            />
            <div
              class="sc-pNWdM kVAvoZ"
            >
              State of Emergency As Hurricane Henri Approaches New York
              <div>
                <button
                  class="sc-gKAaRy kTkCxO"
                  type="button"
                >
                  Add to Favorites
                </button>
              </div>
            </div>
          </div>
          <hr />
          <div
            class="sc-fujyAs ddSJEO"
          >
            <img
              alt="img"
              aria-label="MAGICIAN BRANDS HIS OWN FLESH... WITH A PLAYING CARD!"
              src="https://i.ytimg.com/vi/LqiDJh4zxAc/default.jpg"
            />
            <div
              class="sc-pNWdM kVAvoZ"
            >
              MAGICIAN BRANDS HIS OWN FLESH... WITH A PLAYING CARD!
              <div>
                <button
                  class="sc-gKAaRy kTkCxO"
                  type="button"
                >
                  Add to Favorites
                </button>
              </div>
            </div>
          </div>
          <hr />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
